pkg/rule/builder_test.go

```diff
@@ -0,0 +1,28 @@
+package rule_test
+
+import (
+	"net/netip"
+	"testing"
+
+	"github.com/ngrok/firewall_toolkit/pkg/rule"
+)
+
+func TestBuilder(t *testing.T) {
```

Comment:
Please add negative tests as well as rule output validations similar to the tests in rule_test.go.

Reply:
Agreed. Now that we are happy with the API I will add more extensive test coverage.